src/Casts/MoneyDecimalCast.php

```diff
@@ -16,4 +16,22 @@ protected function getFormatter(Money $money)
     {
         return $money->formatByDecimal();
     }
+
+    public function get($model, string $key, $value, array $attributes)
+    {
+        if (filter_var($value, FILTER_VALIDATE_INT) !== false) {
+            $value = number_format($value, '2');
+        }
+
+        return parent::get($model, $key, $value, $attributes);
+    }
+
+    public function set($model, string $key, $value, array $attributes)
+    {
+        if (filter_var($value, FILTER_VALIDATE_INT) !== false) {
+            $value = number_format($value, '2');
```

Comment:
Not all currencies work with two decimal places

Reply:
Agree, good point. I am unaware of a way to configure casts properly. Could be the same as the currency affix on the cast definition?

Reply:
Given this information: https://docs.adyen.com/development-resources/currency-codes I'd guess that 2 should be fine. Especially when thinking the given case: Here we have an integerish float (like `10.00`) so it's only about forcing the following routines to parse the float value properly.